tests/libgit2/iterator/index.c

```diff
@@ -280,12 +280,24 @@ static const char *expected_index_ci[] = {
 void test_iterator_index__case_folding(void)
 {
 	git_str path = GIT_STR_INIT;
+	git_index *index;
 	int fs_is_ci = 0;
+	int index_is_ci = 0;
+
+	g_repo = cl_git_sandbox_init("icase");
 
-	cl_git_pass(git_str_joinpath(&path, cl_fixture("icase"), ".gitted/CoNfIg"));
+	/* check filesystem case-sensitivity in the sandbox where tests will be executing */
+	cl_git_pass(git_str_joinpath(&path, git_repository_workdir(g_repo), ".git/CoNfIg"));
 	fs_is_ci = git_fs_path_exists(path.ptr);
 	git_str_dispose(&path);
 
+	/* verify index capability matches filesystem detection */
+	cl_git_pass(git_repository_index(&index, g_repo));
+	index_is_ci = (git_index_caps(index) & GIT_INDEX_CAPABILITY_IGNORE_CASE) != 0;
+	cl_assert_equal_i(fs_is_ci, index_is_ci);
+	git_index_free(index);
+	cl_git_sandbox_cleanup();
```

Comment:
Do we need this? We do a `sandbox_cleanup` in the global test `cleanup` function. 🤔

Reply:
I added this cleanup because `index_iterator_test()` below calls its own `cl_git_sandbox_init()` internally. I wanted there to be symmetry, i.e. for the init to start from the same state on every call to `index_iterator_test()`.

That's because the test already calls `cl_git_sandbox_cleanup()` after each call to `index_iterator_test()`, save for the last one which as you point out is cleaned up by `test_iterator_index__cleanup()`.

Reply:
Aha; thanks for the elucidation!